src/telemetry/telemetry.c

```diff
@@ -445,6 +449,46 @@ add_function_call_telemetry(JsonbParseState *state)
 	pushJsonbValue(&state, WJB_END_OBJECT, NULL);
 }
 
+static void
+add_replication_telemetry(JsonbParseState *state)
```

Comment:
I suggest separating the functionality of gathering the stats and updating the JSON instead of putting this all in one function. For instance, defining a struct with the fields we'd like to gather and then putting that in the json or at least reading the data into data fields first and always adding the info at the end.

Now there are several places where this function can exit early, without adding any json fields or just adding some of them. Is this what we want?